ovos_solver_openai_persona/api.py

```
@@ -0,0 +1,177 @@
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": "Bearer " + self.key
+        }
+
+        payload = self._get_common_payload(messages, model)
+
+        try:
+            response = s.post(self.url, headers=headers, data=json.dumps(payload)).json()
+        except json.JSONDecodeError:
+            # TODO: Handle non-JSON responses (e.g. 500 Bad Gateway HTML) gracefully
+            raise RequestException("Failed to decode API response.")
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)